### DISCUSSION TITLE
Support for the GPT-NeoX family of models

### DISCUSSION THREAD

**Discussion by JD-The-65th** [ggml-org/llama.cpp#928] [Ideas]
I think support for the GPT-NeoX models would be nice for a project like this, or as a fork of this project.
[Together AI's Open Source Chatbot is built off of NeoX,](https://huggingface.co/togethercomputer/GPT-NeoXT-Chat-Base-20B) and the [ base Pythia family of models is looking pretty promising too,](https://huggingface.co/EleutherAI/pythia-12b) with newer models being built off of Pythia [(Databricks' Dolly)](https://huggingface.co/databricks/dolly-v2-12b) AFAIK, the model architecture shouldn't be that far off from GPT-J, but take that with a pinch of salt.

**Comment by jploski** [comment]
FWIW, I tried converting https://huggingface.co/OpenAssistant/oasst-sft-1-pythia-12b to ggml today (using the commit 3173a62e), but no luck:

```
!python3 llama.cpp/convert-pth-to-ggml.py oasst-sft-1-pythia-12b 1
Loading model file oasst-sft-1-pythia-12b/pytorch_model-00001-of-00003.bin
Traceback (most recent call last):
  File "/content/llama.cpp/convert-pth-to-ggml.py", line 11, in <module>
    convert.main(['--outtype', 'f16' if args.ftype == 1 else 'f32', '--', args.dir_model])
  File "/content/llama.cpp/convert.py", line 1128, in main
    model_plus = load_some_model(args.model)
  File "/content/llama.cpp/convert.py", line 1055, in load_some_model
    models_plus.append(lazy_load_file(path))
  File "/content/llama.cpp/convert.py", line 849, in lazy_load_file
    return lazy_load_torch_file(fp, path)
  File "/content/llama.cpp/convert.py", line 721, in lazy_load_torch_file
    model = unpickler.load()
  File "/content/llama.cpp/convert.py", line 710, in find_class
    return self.CLASSES[(module, name)]
KeyError: ('torch', 'ByteStorage')
```

**Comment by ggerganov** [comment]
Minimal GPT-NeoX inference example is now available in the `ggml` repo:

https://github.com/ggerganov/ggml/tree/master/examples/stablelm

**Reply by JD-The-65th** [reply]
Nice, thanks a lot man! This project is a really cool one, do you have any way I could financially support you? I'm thinking about giving some money to some cool open source stuff I like.
